graphql_compiler/tests/test_macro_schema.py

```diff
@@ -32,3 +37,115 @@ def test_get_schema_with_macros_basic(self):
             'Animal').fields['out_Animal_RelatedFood'].type
         self.assertTrue(isinstance(related_food_target_type, GraphQLList))
         self.assertEqual('Food', related_food_target_type.of_type.name)
+
+    def test_get_schema_for_macro_definition_addition(self):
+        original_schema = self.macro_registry.schema_without_macros
+        macro_definition_schema = get_schema_for_macro_definition(original_schema)
+        for directive in DIRECTIVES_REQUIRED_IN_MACRO_EDGE_DEFINITION:
+            self.assertTrue(directive in macro_definition_schema.get_directives())
+
+    def test_get_schema_for_macro_definition_retain(self):
+        original_schema = self.macro_registry.schema_without_macros
+        macro_definition_schema = get_schema_for_macro_definition(original_schema)
+        for directive in original_schema.get_directives():
+            if directive in DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINITION:
+                self.assertTrue(directive in macro_definition_schema.get_directives())
+
+    def test_get_schema_for_macro_definition_removal(self):
+        schema_with_macros = get_schema_with_macros(self.macro_registry)
+        macro_definition_schema = get_schema_for_macro_definition(schema_with_macros)
+        for directive in macro_definition_schema.get_directives():
+            self.assertTrue(directive.name != '@output')
+            self.assertTrue(directive.name != '@output_source')
```

Comment:
`directive.name` is a `Name` object. You can get it's string contents by calling `directive.name.value`. If you cargo-culted this code from somewhere else, please check that we are not making the same mistake there too :) It's possible.

Reply:
directive.name is a str. Check again.

Reply:
Ah, correct. It's only a `Name` in directive instances.